src/sif/autocost.cc

```diff
@@ -1114,27 +1066,83 @@ bool HOVCost::AllowedReverse(const baldr::DirectedEdge* edge,
+ * when trying data that may have incorrect restrictions or oneway information.
+ */
+class AutoDataFix : public AutoCost {
+public:
+  /**
+   * Construct auto data fix costing.
+   * Pass in configuration using property tree.
+   * @param  pt  Property tree with configuration/options.
+   */
+  AutoDataFix(const boost::property_tree::ptree& pt) : AutoCost(pt) {
+  }
 
-  if ((edge->forwardaccess() & kHOVAccess) && !(edge->forwardaccess() & kAutoAccess)) {
-    factor *= kHOVFactor;
+  virtual ~AutoDataFix() {
   }
 
-  float sec = (edge->length() * speedfactor_[edge->speed()]);
-  return Cost(sec * factor, sec);
-}
+  /**
+   * Checks if access is allowed for the provided directed edge.
+   * This is generally based on mode of travel and the access modes
+   * allowed on the edge. However, it can be extended to exclude access
+   * based on other parameters such as conditional restrictions and
+   * conditional access that can depend on time and travel mode.
+   * @param  edge           Pointer to a directed edge.
+   * @param  pred           Predecessor edge information.
+   * @param  tile           Current tile.
+   * @param  edgeid         GraphId of the directed edge.
+   * @param  current_time   Current time (seconds since epoch). A value of 0
+   *                        indicates the route is not time dependent.
+   * @param  tz_index       timezone index for the node
+   * @return Returns true if access is allowed, false if not.
+   */
+  virtual bool Allowed(const baldr::DirectedEdge* edge,
+                       const EdgeLabel& pred,
+                       const baldr::GraphTile*& tile,
+                       const baldr::GraphId& edgeid,
+                       const uint32_t current_time,
+                       const uint32_t tz_index) const {
+    // Check access and return false (not allowed if no auto access is allowed in either
+    // direction. Also disallow simple U-turns except at dead-end nodes.
+    if (!((edge->forwardaccess() & kAutoAccess) || (edge->reverseaccess() & kAutoAccess)) ||
+        (!pred.deadend() && pred.opp_local_idx() == edge->localedgeidx()) ||
+        edge->surface() == Surface::kImpassable || IsUserAvoidEdge(edgeid) ||
+        (!allow_destination_only_ && !pred.destonly() && edge->destonly())) {
+      return false;
+    }
+    return true;
+  }
 
-// Check if access is allowed at the specified node.
-bool HOVCost::Allowed(const baldr::NodeInfo* node) const {
-  return (node->access() & kHOVAccess);
-}
+  /**
+   * Returns a function/functor to be used in location searching which will
+   * exclude and allow ranking results from the search by looking at each
+   * edges attribution and suitability for use as a location by the travel
+   * mode used by the costing method. Function/functor is also used to filter
+   * edges not usable / inaccessible by auto in either direction.
+   */
+  virtual const EdgeFilter GetEdgeFilter() const {
```

Comment:
wondering if we need to override the `NodeFilter` as well if we want a truly  no-rules auto costing